pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -188,6 +215,80 @@ def _primitive_bind_call(cls, mps, wires, id=None):
             return type.__call__(cls, mps=mps, wires=wires, id=id)  # pragma: no cover
         return cls._primitive.bind(*mps, wires=wires, id=id)
 
+    def decomposition(self):  # pylint: disable=arguments-differ
+        filtered_hyperparameters = {
+            key: value for key, value in self.hyperparameters.items() if key != "input_wires"
+        }
+        return self.compute_decomposition(
+            self.parameters, wires=self.hyperparameters["input_wires"], **filtered_hyperparameters
+        )
+
+    @staticmethod
+    def compute_decomposition(mps, wires, work_wires):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a product of other operators.
+        The decomposition follows Eq. (23) in `[arXiv:2310.18410] <https://arxiv.org/pdf/2310.18410>`_.
+
+        Args:
+            mps (List[Array]):  list of arrays of rank-3 and rank-2 tensors representing an MPS state as a
+                product of site matrices. See the usage details section for more information.
+
+            wires (Sequence[int]): wires that the template acts on
+            work_wires (Sequence[int]): list of extra qubits needed. The bond dimension of the mps
+                is defined as ``2^len(work_wires)``
+
+        Returns:
+            list[.Operator]: Decomposition of the operator
+        """
+
+        if work_wires is None:
+            raise AssertionError(
+                "The qml.MPSPREP decomposition requires `work_wires` to be specified, "
+                "and the bond dimension cannot exceed `2**len(work_wires)`."
+            )
+
+        ops = []
+        n_wires = len(work_wires) + 1
+
+        for i, Ai in enumerate(mps):
+            vectors = []
+
+            # encodes the tensor Ai in a unitary matrix following Eq.23 in https://arxiv.org/pdf/2310.18410
+            if i == 0:
+                Ai = Ai.reshape((1, *Ai.shape))
+            elif i == len(mps) - 1:
+                Ai = Ai.reshape((*Ai.shape, 1))
```

Comment:
Could we do this `rank-2` to `rank-3` promotion outside the loop itself to prevent checking these conditions for every iteration?

Reply:
Conditional not needed, agree.